app/uk/gov/hmrc/gform/sharedmodel/graph/GraphDataCacheService.scala

```
@@ -0,0 +1,134 @@
+    val (graph, inExprs) = DependencyGraph.toGraph(formModelInterim, AllFormTemplateExpressions(formTemplate))
+    val changedCacheValues = mutable.Map[(DataSource, String), Boolean]()
+
+    val inExprResolverFtr = {
+      val setOfFutures = inExprs.collect { case In(formCtx: FormCtx, dataSource) =>
+        val modelComponentId = formCtx.formComponentId.baseComponentId
+        def makeCall(value: String): Future[Boolean] = dataSource match {
+          case DataSource.Mongo(collectionName) => dbLookupCheckStatus(value, collectionName, hc)
+          case DataSource.Enrolment(serviceName, identifierName) =>
+            retrievals.enrolmentExists(serviceName, identifierName, value).pure[Future]
+          case dd @ DataSource.DelegatedEnrolment(_, _) =>
+            retrievals.maybeGovermentGatewayId.fold(false.pure[Future]) { governmentGatewayId =>
+              delegatedEnrolmentCheckStatus(governmentGatewayId, dd, IdentifierValue(value), hc)
+            }
+          case DataSource.SeissEligible =>
+            seissEligibilityChecker(UtrEligibilityRequest(value), hc)
+        }
+
+        variadicFormData.forBaseComponentId(modelComponentId).map { case (modelComponentId, variadicFormValue) =>
+          val value = variadicFormValue.fold(one => one.value)(many =>
+            throw new RuntimeException(s"Didn't expect many data points in data $many")
+          )
+
+          val result = booleanExprCache.get(dataSource, value).map(Future.successful).getOrElse {
+
+            makeCall(value).map { result =>
+              changedCacheValues.addOne((dataSource -> value, result))
+              result
+            }
+          }
+
+          val newIn = In(new FormCtx(modelComponentId.toFormComponentId), dataSource)
+          result.map(newIn -> _)
+        }
+      }.flatten
+
+      Future
+        .sequence(setOfFutures)
+        .map(_.toMap)
+        .map { map => (in: In) =>
+          map.getOrElse(in, false)
```

Can we throw an exception in case `in` is not in cache?

With current solution not all `in` expression are evaluated so either we do something about it or we throw an exception.

Btw. this is not introduced by this PR. This problem exists in `main` too.

This is the case which should throw an exception:
```
{
  "type": "info",
  "id": "id1",
  "label": "",
  "infoType": "noformat",
  "infoText": "Your VAT number is ${if vatNumber in mongo.mtdVatNumber then 'FOO' else 'BAR'}"
}
```

(assuming `vatNumber in mongo.mtdVatNumber` is not used in `includeIf`)

Unless you want to fix it now, we will fix this properly later in other ticket.